Paper: Spatial Microsimulation and Activity Allocation in Python: An Update on the Likeness Toolkit

jGaboardi opened the pull request:
Unfortunately, we needed to submit late due to a significant delay in our internal review process. Many apologies.

---------------

If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@mepa<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@ameyxd, @kalyan678<!--end-reviewers-list-->


cc @jvtcl

Commit:
initial commit

Commit:
Merge remote-tracking branch 'upstream/dev' into 2023

Commit:
Merge branch '2023' into 2023_tuccardi_paper

Commit:
Merge remote-tracking branch 'upstream/dev' into 2023

Commit:
Merge branch '2023' into 2023_tuccardi_paper

Commit:
run first draft -passing locally

Commit:
Merge remote-tracking branch 'upstream/2023' into 2023

Commit:
Merge branch '2023' into 2023_tuccardi_paper

Commit:
Merge remote-tracking branch 'upstream/2023' into 2023

Commit:
Merge branch '2023' into 2023_tuccardi_paper

jGaboardi commented:
@scoobies build status

scoobies commented:
{"jGaboardi-2023_tuccardi_paper"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmpdtwrueod'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: james_gaboardi\n*** Warning: /tmp/tmpdtwrueod/publisher/../output/james_gaboardi/paper_stats.json does not exist.\n*** Warning: /tmp/tmpdtwrueod/publisher/_build/toc.json does not exist.\n*** Warning: /tmp/tmpdtwrueod/publisher/_build/other.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/jGaboardi-2023_tuccardi_paper.pdf", "build_status"=>"success", "build_timestamp"=>"06/06 15:16"}, "status"=>"success"}}

jGaboardi commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.5281/zenodo.7843975 is OK
- 10.5281/zenodo.7768810 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1147/rd.471.0057 is OK
- 10.1016/j.compenvurbsys.2013.03.004 is OK
- 10.1111/gec3.12204 is OK
- 10.25080/majora-212e5952-014 is OK
- 10.25080/majora-212e5952-02d is OK
- 10.6084/m9.figshare.c.6139131.v2 is OK

MISSING DOIs

- 10.1016/j.epidem.2022.100632 may be a valid DOI for title: Assessing K-12 school reopenings under different COVID-19 Spread scenarios–United States, school year 2020/21: A retrospective modeling study
- 10.1109/wsc.2018.8632409 may be a valid DOI for title: Chisim: An agent-based simulation model of social interactions in a large urban area
- 10.1109/jstars.2018.2835377 may be a valid DOI for title: Building extraction at scale using convolutional neural network: Mapping of the United States

INVALID DOIs

- https://doi.org/10.1111/gean.12276 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.apgeog.2022.102844 is INVALID because of 'https://doi.org/' prefix
- 10.2172/1868508 is INVALID
```

Commit:
correct DOIs

Commit:
update DOI + refs [2]

jGaboardi commented:
@scoobies build status

scoobies commented:
{"jGaboardi-2023_tuccardi_paper"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmp7sexlv4i'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: james_gaboardi\n*** Warning: /tmp/tmp7sexlv4i/publisher/../output/james_gaboardi/paper_stats.json does not exist.\n*** Warning: /tmp/tmp7sexlv4i/publisher/_build/toc.json does not exist.\n*** Warning: /tmp/tmp7sexlv4i/publisher/_build/other.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/jGaboardi-2023_tuccardi_paper.pdf", "build_status"=>"success", "build_timestamp"=>"06/06 16:04"}, "status"=>"success"}}

jGaboardi commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.5281/zenodo.7843975 is OK
- 10.5281/zenodo.7768810 is OK
- 10.1016/j.epidem.2022.100632 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1147/rd.471.0057 is OK
- 10.1016/j.compenvurbsys.2013.03.004 is OK
- 10.1109/WSC.2018.8632409 is OK
- 10.1111/gec3.12204 is OK
- 10.1111/gean.12276 is OK
- 10.25080/majora-212e5952-014 is OK
- 10.25080/majora-212e5952-02d is OK
- 10.1016/j.apgeog.2022.102844 is OK
- 10.1109/JSTARS.2018.2835377 is OK
- 10.6084/m9.figshare.c.6139131.v2 is OK

MISSING DOIs

- None

INVALID DOIs

- 10.2172/1868508 is INVALID
```

jGaboardi commented:
> INVALID DOIs
> - 10.2172/1868508 is INVALID

`10.2172/1868508` is the correct DOI for *Foundational Science for Biopreparedness and Response*. See the bottom of page 2 [here](
https://science.osti.gov/-/media/Initiatives/pdf/Bioprepardeness_Roundtable_Report_092722.pdf).

deniederhut commented:
Haha I believe you

deniederhut commented:
@scoobies assign @mepa as editor

scoobies commented:
Assigned! @mepa is now the editor

deniederhut commented:
@scoobies add @ameyxd to reviewers

scoobies commented:
@ameyxd added to the reviewers list!

deniederhut commented:
@scoobies code of conduct

scoobies commented:
Our CoC: https://www.scipy2023.scipy.org/code-of-conduct

scoobies commented:
Reports of abusive or harassing behavior may be reported to scipy@enthought.com

deniederhut commented:
@scoobies add @kalyan678 to reviewers

scoobies commented:
@kalyan678 added to the reviewers list!

mepa commented:
Hey @ameyxd and @kalyan678! A friendly reminder that initial reviews are due Friday, June 23rd.

kalyan678 commented on papers/james_gaboardi/metadata.yaml:
```diff
@@ -0,0 +1,20 @@
+abstract:
+  - "Understanding human security and social equity issues within human systems requires large-scale models of population dynamics that simulate high-fidelity representations of individuals and access to essential activities (work/school, social, errands, health). Likeness is a Python toolkit that provides these capabilities for Oak Ridge National Laboratory's (ORNL) UrbanPop spatial microsimulat ...
```
The abstract is well-written and informative. It provides a clear overview of the paper's purpose, methods, and findings. The only suggestion I would make is to break up the first sentence into two sentences, for better readability.

For example: 
- Understanding human security and social equity issues requires large-scale models of population dynamics that simulate high-fidelity representations of individuals and access to essential activities. Likeness is a Python toolkit that provides these capabilities for Oak Ridge National Laboratory's (ORNL) UrbanPop spatial microsimulation project.

- In this paper, we describe the expansion and scaling of Likeness capabilities to metropolitan areas in the United States. We then provide an integrated demonstration of our methods based on a case study of Leon County, FL. We perform validation exercises on 1) neighborhood demographic composition and 2) visits by demographic cohorts (gender/age) obtained from point of interest (POI) footfall data for essential services (grocery stores).

jvtcl replied:
- [x] @jvtcl to address this.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\subsection{Introduction}\label{introduction}
+
+Agent-based models linking individual behavior to population dynamics are essential for understanding human security and social equity issues within human systems \cite{germann2022assessing, ozik2021population, urbanpop-AG-2023}. A current challenge for these models is to more directly account for population heterogeneity \cite{BRaVE_DOE_2022}, which relates to factors including socio-demographic ...
```
The introduction of the paper provides a clear overview of the research topic and highlights the significance of agent-based models in understanding human security and social equity issues. Here are some suggestions to improve the different aspects of the introduction:

- The introduction is well-written and effectively communicates the main points. However, you could enhance the clarity and flow by breaking down the long sentences into shorter ones. Additionally, consider using more specific and concise language to avoid unnecessary repetition.
- Lastly, in the first sentence, you could add a bit more detail about what agent-based models are and why they are important. For example, you could say something like "Agent-based models are computer simulations that allow us to model the behavior of individual agents and how their interactions can give rise to complex system-level behavior."

jvtcl replied:
- [x] @jvtcl to address this.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+Agent-based models linking individual behavior to population dynamics are essential for understanding human security and social equity issues within human systems \cite{germann2022assessing, ozik2021population, urbanpop-AG-2023}. A current challenge for these models is to more directly account for population heterogeneity \cite{BRaVE_DOE_2022}, which relates to factors including socio-demographic ...
+
+Foundational examples produced for Likeness have explored travel routing from home to anchor activities (e.g., work, school), POI occupancy characteristics \cite{likeness-scipy-paper-2022, likeness-scipy-poster-2022}, as well as routing incidental travel from anchor to non-anchor activities (e.g., social, errands, health) \cite{likeness_aag_2023}. This paper builds from these foundational example ...
```
Consider highlighting particular societal or scientific issues that can be solved using the developed methodologies, in my opinion, to further emphasize the paper's significance. Be sure to emphasize the insights that may be gained from the combination of synthetic populations, real-world data, and activity allocation in a variety of fields, including urban planning, public health, and disaster preparedness. I think this will help readers to find the introduction more interesting, informative, and appealing.

ameyxd replied:
I agree with this. The work is technically sound, but will benefit from highlighting the more obvious implications and insights to appeal to the general audience. @jGaboardi you can perhaps attend to this at the end of your abstract, Introduction and/or conclusion.

jvtcl replied:
- [x] @jvtcl to address this.

kalyan678 commented:
> Hey @ameyxd and @kalyan678! A friendly reminder that initial reviews are due Friday, June 23rd.

Hey @mepa - I appreciate the reminder. On it, I already am. Cheers!

ameyxd commented:
@mepa thanks for reminding. @jGaboardi happy to review 😄

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+In step with the creation of foundational examples for Likeness, we have developed an integrated workflow that scales the ecosystem (Figure~\ref{fig:ecosystem}) to support microsimulation for any metropolitan statistical area (MSA) in the United States.
+
+Figure~\ref{fig:parse-to-attribution} outlines the procedure for agent generation at the MSA level. As discussed in \cite{likeness-scipy-paper-2022}, the \texttt{livelike.acs.puma} class is the core object for residential population synthesis, which stores all census microdata and geographic (e.g., block group, tract) model constraints to support spatial allocation for a single Public-Use Microda ...
```
- I think it would be good to provide more details about the validation exercises that conducted. For example, You could discuss the specific metrics that they used to measure the accuracy of the simulations, and they could provide more information about the results of the validation exercises.

jvtcl replied:
- [x] @jvtcl to add a citation for CCA [here] (https://github.com/jGaboardi/scipy_proceedings/blob/f49ce084823d019f9c7458534a7211078f99f2c4/papers/james_gaboardi/paper.tex#L68).

@kalyan678 the highlighted text does not seem to match the section that you'd like us to update. Could you please point us to the lines in the text that need revision?

kalyan678 replied:
@jvtcl - Sorry for the confusion. I completely misplaced my comment.
I think you made a point in the abstract and on page 5 of the study about conducting validation exercises. These were the lines I was referring to. (We were also
limited to only one month of POI visit data, but we hope
to expand its scope for future validation exercises)

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+Agent generation at residential locations via \texttt{livelike} provides origin points for simulating travel to essential activities. 
+The next stage in the workflow employs network analysis to model the cost of travel to these activities \cite{OpenStreetMap, osmnx_CEUS_2017, foti_generalized_2012} and allocate agents to POIs accordingly based on mathematical programming routines \cite{mitchell_pulp_2011, santos_mixed_2020, lougee_coin_2003, forrest_coinorcbc_2023}. In our first iteration of Likeness, both these tasks were accom ...
+
+Finally -- and at the heart of it all -- expansion and scaling of the Likeness ecosystem led to the development of a new package for common utilities, \texttt{likeness-vitals}, which provides support for monitoring and timing processes, data manipulation, shared spatial functionality, and Census API access.
```
- The future use of the Likeness toolset has been outlined in clear terms, and I appreciated the thorough citations of pertinent academic works.
- Overall, I believe that we can simplify the language and sentence structure to make the content more accessible to readers. Break down complex sentences into smaller, more concise ones.

jvtcl replied:
- [x] @jvtcl to address this.

jvtcl replied:
I made a runthrough of the manuscript and broke up sentences where possible.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\subsection{Integrated Demonstration: Leon County, Florida} \label{section:LCFL}
+
+Following the workflow described in Section~\nameref{section:likeness-expansion-scaling}, we demonstrate the current capabilities of Likeness and validate our activity allocation routine for Leon County, Florida. Leon County, whose primary city is Tallahassee, features a population of just under 300,000 residents, a compact urban footprint, and a diverse array of transportation modes (driving, tr ...
```
- Although you provided a reference to TransitFeeds data, I double-checked and discovered that "TransitFeeds data is not regularly updated. Find up-to-date GTFS Schedule data on the the Mobility Database catalogs." I do not see any such highlight in the paper.

jvtcl replied:
- [x] @jGaboardi to address this.

jGaboardi replied:
@kalyan678 

In that sentence we also provide a comment on The Mobility Database Catalogs and a footnote including a link to the GitHub Repo (should be footnote (8.)). Is there something more you are looking for here?

> The GTFS datasets can be obtained via services such as *TransitFeeds*[^7] and *The Mobility Database Catalogs*[^8].

> [^7]: https://transitfeeds.com/
> [^8]: https://github.com/MobilityData/mobility-database-catalogs

kalyan678 replied:
@jGaboardi Thank you for your clarification. I've seen the specifics, and I believe we're in good hands here :)

jGaboardi replied:
@jvtcl Please mark as completed.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+Following the workflow described in Section~\nameref{section:likeness-expansion-scaling}, we demonstrate the current capabilities of Likeness and validate our activity allocation routine for Leon County, Florida. Leon County, whose primary city is Tallahassee, features a population of just under 300,000 residents, a compact urban footprint, and a diverse array of transportation modes (driving, tr ...
+
+We first simulated a single synthetic population for the Tallahassee Core-Based Statistical Area (CBSA), based on constraining variables including technical characteristics linked to sampling universe totals (i.e., population, housing units, households), as well as descriptive factors including demographics, socioeconomic status, housing, mobility, and worker and student characteristics. For the  ...
```
There are a few technical terms and acronyms that could be briefly explained or replaced with simpler language to improve understanding for a broader audience. For example, instead of "PUMA 1206300," you could mention that it refers to a specific geographical area.

ameyxd replied:
I disagree, perhaps add geographical area but don’t remove the PUMA. It is descriptive and helps people familiar with the topic quickly identify the region if they want to.

jvtcl replied:
- [x] @jvtcl to clarify this.

kalyan678 replied:
I apologise for the confusion. It is not my intention to remove the PUMA. The goal is to add the geographical area rather than the code.

jvtcl replied:
Added the [PUMA name](https://censusreporter.org/profiles/79500US1206300-apalachee-region-outside-leon-county-puma-fl/) as provided by Census.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\subsection{Results}\label{section:results}
+
+\subsubsection{Demographic Validation}
```
While the text provides detailed information about the methodology, it could benefit from providing a bit more context about the significance and real-world implications of the study. Explain why understanding travel behavior and infrastructure planning is important for urban areas like Leon County.

jvtcl replied:
- [x] @jvtcl to address this.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\input figures/local_demographic_validation.tex
+
+Each of the three P-MEDM runs (one for each PUMA in the Tallahassee CBSA) resulted in a synthetic reconstruction of the ACS 90\% MOEs that conformed within 99\% of the published MOEs from the ACS SF (Table~\ref{tab:demographic-validation}. Mapped to Leon County block groups (Figure~\ref{fig:local-demographic-validation}), MOE Fit Rates were somewhat lower but still conform within 90\% of the ACS  ...
```
- It was difficult to grasp the main point here. My point here is break them down into shorter, more digestible sentences. For example, instead of "Each of the three P-MEDM runs (one for each PUMA in the Tallahassee CBSA) resulted in a synthetic reconstruction of the ACS 90% MOEs that conformed within 99% of the published MOEs from the ACS SF," you could say "Each of the three P-MEDM runs (one for each PUMA in the Tallahassee CBSA) resulted in a synthetic reconstruction of the ACS 90% MOEs. These reconstructions conformed within 99% of the published MOEs from the ACS SF." Let me know how that sounds?

jvtcl replied:
@kalyan678 We agree with this suggestion. 

- [x] @jvtcl to address this.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\input figures/local_win_cca.tex
+
+Our mobility simulation more realistically recreated between-destination demographics ($R^2$ = 0.74) than within-destination demographics ($R^2$ = 0.41) (Table~\ref{tab:demographic-validation}). Inspecting the local within-destination scores, mapped in Figure~\ref{fig:local-win-cca}, we observed generally greater correspondence between synthetic and observed POI demographics near Tallahassee's do ...
```
- It would be really helpful if you can provide more details about the limitations and challenges faced during the mobility validation exercise and the modest success achieved. This will help readers understand the context and potential areas for improvement.

jvtcl replied:
- [x] @jvtcl to address this.

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+    \item \textbf{Tooling for custom geographic extents.} The MSA-specific workflow demonstrated in this paper is limited in that it does not support custom geographic extents, limiting analysis, for example, of predominantly rural areas. 
+    We are actively developing an approach to create residential synthetic populations for custom areas of interest (AOIs), supported by USA Structures, which will include national-scale coverage. 
+    \item \textbf{Open-sourcing core packages.} Though we have yet to meet our goal of open-sourcing the suite of Likeness packages \cite{likeness-scipy-paper-2022}, we are still on track to release the core packages for residential population synthesis, \texttt{pymedm} and \texttt{livelike}, in 2023, with releases of \texttt{movelike} and \texttt{actlike} likely to follow in 2024.
+    \item \textbf{Packaging schema.} A further consideration related to open-sourcing is whether we should migrate from a confederated toolkit schema where each module is a semi-independent Python package, as is seen in the modern implementation of PySAL \cite{pysal_GA_2022}, to a single monolithic Python package with submodules. Each schema has benefits and this decision will require much consid ...
```
May be, it's good to elaborate on the benefits and considerations associated with migrating from a confederated toolkit schema to a monolithic Python package. Discuss the implications and potential advantages of each approach.

jvtcl replied:
- [x] @jGaboardi to address this.

jGaboardi replied:
@jvtcl I have added some more explanatory language in for this. Please mark my portion as completed.

kalyan678 commented on papers/james_gaboardi/figures/ecosystem.tex:
```diff
@@ -0,0 +1,6 @@
+\centering
+\fbox{\includegraphics[width=150mm]{figures/ecosystem.pdf}}
+\caption{Likeness ecosystem overview.}
+\label{fig:ecosystem}
```
The Likeness ecosystem overview is pretty straightforward!

kalyan678 commented on papers/james_gaboardi/figures/mode_assign_workflow.tex:
```diff
@@ -0,0 +1,6 @@
+\centering
+\fbox{\includegraphics[width=150mm]{figures/mode_assign_workflow.png}}
+\caption{Agent travel mode assignment decision tree.}
+\label{fig:mode_assign_workflow}
```
Decision tree is visually appealing and easy to follow. Keep up the good work!!

kalyan678 commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\bibliographystyle{ieeetr}
+\bibliography{ mybib }
+
+\end{document}
```
Overall, the paper is well-written and informative. The quality of the writing is good. The content is well-structured and organized, making it easy to follow the flow of information. However, I feel some sentences and phrases could be simplified to enhance readability. Finally, the paper demonstrates a strong technical understanding and presents a detailed analysis of the research. It effectively communicates the objectives, methods, and results of the study.

jvtcl replied:
@kalyan678 We appreciate your comments and will address readability and discussion points you mentioned in your review.

kalyan678 commented:
Hello Everyone - I apologies for introducing myself so late. I'm sorry about that.
Kalyan, a self-taught data scientist, analytics manager, and open-source enthusiast, has graced prestigious stages worldwide with his talks. He's also a valued mentor and reviewer for renowned conferences and hackathons, and actively contributes to the growth of open source communities.


Dear @mepa  @deniederhut - Thank you for giving me the opportunity to review the scientific paper. @jGaboardi - It was a pleasure to read and evaluate your work. I have thoroughly reviewed the paper and have added my comments accordingly.

I appreciate the efforts you have put into enhancing the Likeness toolkit and demonstrating its capabilities through the mobility validation exercise. Your research raises interesting questions and presents potential avenues for further development.

I have provided detailed comments and suggestions throughout the paper. Please refer to them and let me know if you have any questions. I hope you find my feedback helpful in advancing your research.

Once again, thank you for sharing your work with me. I look forward to any further discussions or iterations on the paper.

This brings my initial review to the end. Once again, many thanks!

Cheers
Kalyan

jGaboardi commented:
@kalyan678 Thank you completing your review! Perhaps you forgot to post your actual review comments?

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\subsection{Introduction}\label{introduction}
+
+Agent-based models linking individual behavior to population dynamics are essential for understanding human security and social equity issues within human systems \cite{germann2022assessing, ozik2021population, urbanpop-AG-2023}. A current challenge for these models is to more directly account for population heterogeneity \cite{BRaVE_DOE_2022}, which relates to factors including socio-demographic ...
```
A clear definition or equivalent term for 'agent' would be appreciated in your introduction/overview. Perhaps mentioning 'agents' are 'simulated individuals' - could be obvious for the authors but not the reader.

ameyxd replied:
A similar case can be made for another keyword - 'population dynamics' - to convey the purpose.

jvtcl replied:
- [x] @jvtcl to address this

jvtcl replied:
- Generalized "agent-based model" to "synthetic population" (and removed "population synthesis")
- Changed "human dynamics" -> "population dynamics" to match the text
- Added "microsimulation"

ameyxd commented on papers/james_gaboardi/mybib.bib:
```diff
@@ -0,0 +1,315 @@
+                  Tuccillo, Joseph Vincent},
+  title        = {{Spatial Microsimulation and Activity Allocation 
+                   for Examining COVID-19 Vaccine Access Profiles}},
+  month        = {mar},
```
mar -> Mar

A general recommendation to check for and maintain consistency in month naming and periods after

jvtcl replied:
- [x] @jGaboardi to address this

jGaboardi replied:
@jvtcl This is completed locally in our ShareLeaf. Please mark as done. (I can't tick off your check box)

ameyxd commented on papers/james_gaboardi/mybib.bib:
```diff
@@ -0,0 +1,315 @@
+	urldate = {2022-05-05},
+	journal = {Computers, Environment and Urban Systems},
+	author = {Lovelace, Robin and Ballas, Dimitris},
+	month = sep,
```
sep -> Sept.

jvtcl replied:
- [x] @jGaboardi to address this

jGaboardi replied:
@jvtcl Please mark as completed.

ameyxd commented on papers/james_gaboardi/mybib.bib:
```diff
@@ -0,0 +1,315 @@
+    publisher={figshare},
+    author={Yang, H. Lexie and Tuttle, Mark and Laverdiere, Melanie and Hauser, Taylor and Swan, Benjamin and Schmidt, Erik and Moehl, Jessica and Reith, Andrew and McKee, Jacob and Whitehead, Matt},
+    year={2022},
+    month={Sep}
```
sep -> Sept.

jvtcl replied:
- [x] @jGaboardi to address this

jGaboardi replied:
@jvtcl Please mark as completed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+Agent-based models linking individual behavior to population dynamics are essential for understanding human security and social equity issues within human systems \cite{germann2022assessing, ozik2021population, urbanpop-AG-2023}. A current challenge for these models is to more directly account for population heterogeneity \cite{BRaVE_DOE_2022}, which relates to factors including socio-demographic ...
+
+Foundational examples produced for Likeness have explored travel routing from home to anchor activities (e.g., work, school), POI occupancy characteristics \cite{likeness-scipy-paper-2022, likeness-scipy-poster-2022}, as well as routing incidental travel from anchor to non-anchor activities (e.g., social, errands, health) \cite{likeness_aag_2023}. This paper builds from these foundational example ...
```
I could not find the work associated with likeness_aag_2023. Is this published or currently under review? If so, please indicate in References.

jvtcl replied:
- [x] @jGaboardi to address this. I think there's a Zenodo for this? We might have just forgotten to add the BibTeX.

jGaboardi replied:
We have a Zenodo DOI for it and it is listed in out references section:

> [7] J. D. Gaboardi and J. V. Tuccillo, “Spatial Microsimulation and ActivityAllocation  for  Examining  COVID-19  Vaccine  Access  Profiles,”  Mar.2023.

It is a conference presentation with designated DOI. However, the BiBteX style was omitting the `url` field. I updated the entry with a `howpublished` field that now shows the proper URL.

@jvtcl Please mark as completed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+Agent-based models linking individual behavior to population dynamics are essential for understanding human security and social equity issues within human systems \cite{germann2022assessing, ozik2021population, urbanpop-AG-2023}. A current challenge for these models is to more directly account for population heterogeneity \cite{BRaVE_DOE_2022}, which relates to factors including socio-demographic ...
+
+Foundational examples produced for Likeness have explored travel routing from home to anchor activities (e.g., work, school), POI occupancy characteristics \cite{likeness-scipy-paper-2022, likeness-scipy-poster-2022}, as well as routing incidental travel from anchor to non-anchor activities (e.g., social, errands, health) \cite{likeness_aag_2023}. This paper builds from these foundational example ...
+discuss scaling our methods to larger areas of interest, as well as examines the performance of our activity allocation model relative to real-world POI demographics from digital trace data.
```
Please define or expand on the phrase 'digital trace data'.

jvtcl replied:
- [x] @jvtcl to address this.

jvtcl replied:
I added a brief clarification that this refers to "anonymized mobile device" data.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+Agent-based models linking individual behavior to population dynamics are essential for understanding human security and social equity issues within human systems \cite{germann2022assessing, ozik2021population, urbanpop-AG-2023}. A current challenge for these models is to more directly account for population heterogeneity \cite{BRaVE_DOE_2022}, which relates to factors including socio-demographic ...
+
+Foundational examples produced for Likeness have explored travel routing from home to anchor activities (e.g., work, school), POI occupancy characteristics \cite{likeness-scipy-paper-2022, likeness-scipy-poster-2022}, as well as routing incidental travel from anchor to non-anchor activities (e.g., social, errands, health) \cite{likeness_aag_2023}. This paper builds from these foundational example ...
```
It may help to have a clear distinction between 'anchor' and 'non-anchor' POIs. For example, is 'home' considered anchor as implied from "anchor to non-anchor activities (e.g. social)..."?

jvtcl replied:
@ameyxd We consider home to be separate from anchor activities like school and work. We will clarify by using the terminology "home locations". 

- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\input figures/ecosystem.tex
+
+In step with the creation of foundational examples for Likeness, we have developed an integrated workflow that scales the ecosystem (Figure~\ref{fig:ecosystem}) to support microsimulation for any metropolitan statistical area (MSA) in the United States.
```
I do not recommend starting with a preposition here. "Synchronous with" or similar may be better.

jvtcl replied:
- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/figures/ecosystem.tex:
```diff
@@ -0,0 +1,6 @@
+\begin{figure*}[htbp]
```
Please consider maintaining consistency in font styles used across all figures.

jvtcl replied:
@ameyxd We appreciate this comment, but this may be a challenge for us, as the figures were generated through various means, including PowerPoint, LucidChart, `matplotlib`, as well as Stamen Maps, which all use different font styles.

jGaboardi replied:
Although it is a great suggestion, we were unable to accommodate this due to the varying software packages used for generation (as @jvtcl said before).

ameyxd commented on papers/james_gaboardi/figures/ecosystem.tex:
```diff
@@ -0,0 +1,6 @@
+\begin{figure*}[htbp]
+\centering
+\fbox{\includegraphics[width=150mm]{figures/ecosystem.pdf}}
```
Dotted link from Agent Generation to '3' is undirected, is that missing an arrow? You could consider making it horizontal.

jvtcl replied:
- [x] @jvtcl to address this.

jvtcl replied:
We also need to remove the asterisk from the "Activity Model" bubble. This was from an earlier version of the figure and no longer is needed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+In step with the creation of foundational examples for Likeness, we have developed an integrated workflow that scales the ecosystem (Figure~\ref{fig:ecosystem}) to support microsimulation for any metropolitan statistical area (MSA) in the United States.
+
+Figure~\ref{fig:parse-to-attribution} outlines the procedure for agent generation at the MSA level. As discussed in \cite{likeness-scipy-paper-2022}, the \texttt{livelike.acs.puma} class is the core object for residential population synthesis, which stores all census microdata and geographic (e.g., block group, tract) model constraints to support spatial allocation for a single Public-Use Microda ...
```
FIPS codes are not defined or explained. If relevant, please expand here.

jvtcl replied:
- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\subsection{Integrated Demonstration: Leon County, Florida} \label{section:LCFL}
+
+Following the workflow described in Section~\nameref{section:likeness-expansion-scaling}, we demonstrate the current capabilities of Likeness and validate our activity allocation routine for Leon County, Florida. Leon County, whose primary city is Tallahassee, features a population of just under 300,000 residents, a compact urban footprint, and a diverse array of transportation modes (driving, tr ...
```
This is a general comment for the reference to sections: When using \nameref, try to number sections to reduce redundancy and verbosity, if the paper format allows for it (by setting \documentclass[conference] or by changing \setcounter). This goes for references to sections throughout the document.

jvtcl replied:
- [ ] @jGaboardi to address this.

jGaboardi replied:
@ameyxd We are not sure that the SciPy Proceedings allow for that and we could not find examples of that format in other submissions for this year. Perhaps we can get an editor's opinion on the matter? cc @mepa

jGaboardi replied:
We will retain our `\nameref` format that is compatible with the SciPy proceeding template unless specifically guided to do otherwise.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\input figures/parse-to-attribution
+
+Agent generation at residential locations via \texttt{livelike} provides origin points for simulating travel to essential activities. 
+The next stage in the workflow employs network analysis to model the cost of travel to these activities \cite{OpenStreetMap, osmnx_CEUS_2017, foti_generalized_2012} and allocate agents to POIs accordingly based on mathematical programming routines \cite{mitchell_pulp_2011, santos_mixed_2020, lougee_coin_2003, forrest_coinorcbc_2023}. In our first iteration of Likeness, both these tasks were accom ...
```
Please expand on or mention the programming routines/libraries referenced here: PuLP, MILP, COIN-OR, etc.

jvtcl replied:
- [x] @jGaboardi to address this.

jGaboardi replied:
@jvtcl I expanded on this within a footnote in our internal ShareLeaf.

jGaboardi replied:
@jvtcl ~~please mark complete~~ just realized you already did that.

ameyxd commented on papers/james_gaboardi/mybib.bib:
```diff
@@ -0,0 +1,315 @@
+
+
+@misc{forrest_coinorcbc_2023,
+  author       = {John Forrest and
```
Please update author list with  proper names, or mention [Github profile] when applicable here.

jvtcl replied:
- [x] @jGaboardi to address this.

jGaboardi replied:
All software references in the manuscript are the official recommended ones from respective packages. We do not think a change is needed.

jGaboardi replied:
@jvtcl Please mark as completed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\input figures/parse-to-attribution
+
+Agent generation at residential locations via \texttt{livelike} provides origin points for simulating travel to essential activities. 
+The next stage in the workflow employs network analysis to model the cost of travel to these activities \cite{OpenStreetMap, osmnx_CEUS_2017, foti_generalized_2012} and allocate agents to POIs accordingly based on mathematical programming routines \cite{mitchell_pulp_2011, santos_mixed_2020, lougee_coin_2003, forrest_coinorcbc_2023}. In our first iteration of Likeness, both these tasks were accom ...
```
```suggestion
The next stage in the workflow employs network analysis to model the cost of travel to these activities \cite{OpenStreetMap, osmnx_CEUS_2017, foti_generalized_2012} and allocate agents to POIs accordingly based on mathematical programming routines \cite{mitchell_pulp_2011, santos_mixed_2020, lougee_coin_2003, forrest_coinorcbc_2023}. In our first iteration of Likeness, both these tasks were accomplished within \texttt{actlike}. However, we concluded that the network modeling piece was specialized enough to be split from the \texttt{actlike} package, which led to the creation of \texttt{movelike}. With a push for varied modes of network traversal, three new modes of travel can now be modeled: walking, biking, public transit. However, modeling travel behavior via public transportation is less straightforward than for driving, biking, and walking networks due to stricter network topology, including factors like connectivity and directionality of routes. We have made our foray into modeling more realistic public transit behavior within \texttt{movelike} through the incorporation of the General Transit Feed Specification (GTFS)\footnote{https://gtfs.org/}. GTFS is a data specification that stipulates the required files, along with their structure and format\footnote{https://gtfs.org/schedule/reference/\#dataset-files}, for publishing, ingesting, and utilizing public transit datasets. The GTFS datasets can be obtained via services such as \textit{TransitFeeds}\footnote{https://transitfeeds.com/} and \textit{The Mobility Database Catalogs}\footnote{https://github.com/MobilityData/mobility-database-catalogs}. In our current iteration we utilize GTFS data feeds to implement a pseudo-transit network space by which agents can engage in limited traversal. This is accomplished through a mask of \textit{OpenStreetMap}\footnote{https://www.openstreetmap.org/} (OSM) street segments known to be associated with bus routes. The OSM network is masked by passing a (multi)polygon feature of buffered and unioned bus routes within the study area into \texttt{osmnx} \cite{osmnx_CEUS_2017}. This method demonstrates progression in representing public transit but certainly has room for improvement, which will be discussed in \nameref{section:dev-roadmap}.
```

ameyxd replied:
Grammar and phrasing - 'made our foray', etc. 'first foray' is redundant - foray implies first attempt.

jvtcl replied:
- [ ] @jGaboardi to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\subsection{Integrated Demonstration: Leon County, Florida} \label{section:LCFL}
+
+Following the workflow described in Section~\nameref{section:likeness-expansion-scaling}, we demonstrate the current capabilities of Likeness and validate our activity allocation routine for Leon County, Florida. Leon County, whose primary city is Tallahassee, features a population of just under 300,000 residents, a compact urban footprint, and a diverse array of transportation modes (driving, tr ...
```
```suggestion
Following the workflow described in Section~\nameref{section:likeness-expansion-scaling}, we demonstrate the current capabilities of Likeness and validate our activity allocation routine for Leon County, Florida. Leon County, whose primary city is Tallahassee, features a population of just under 300,000 residents, a compact urban footprint, and a diverse array of transportation modes (driving, transit, bike, walking). Our mobility validation exercise is based on grocery store visits from simulated home locations. Grocery stores provide a useful test case because they are both catchments for the general population and points of access to vital services including food and healthcare. We obtained grocery store visits from Foursquare's Research Visits feed, which provides footfall data attributed by demographic cohort (gender by age) for a variety of POI types\footnote{https://location.foursquare.com/places/docs/how-does-places-work}\textsuperscript{,}\footnote{https://location.foursquare.com/visits/docs/research-feed-schema}. 
```

ameyxd replied:
POI is defined.

jvtcl replied:
- [ ] @jGaboardi to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+Following the workflow described in Section~\nameref{section:likeness-expansion-scaling}, we demonstrate the current capabilities of Likeness and validate our activity allocation routine for Leon County, Florida. Leon County, whose primary city is Tallahassee, features a population of just under 300,000 residents, a compact urban footprint, and a diverse array of transportation modes (driving, tr ...
+
+We first simulated a single synthetic population for the Tallahassee Core-Based Statistical Area (CBSA), based on constraining variables including technical characteristics linked to sampling universe totals (i.e., population, housing units, households), as well as descriptive factors including demographics, socioeconomic status, housing, mobility, and worker and student characteristics. For the  ...
```
```suggestion
We first simulated a single synthetic population for the Tallahassee Core-Based Statistical Area (CBSA), based on constraining variables including technical characteristics linked to sampling universe totals (i.e., population, housing units, households), as well as descriptive factors including demographics, socioeconomic status, housing, mobility, and worker and student characteristics. For the remainder of the analysis, we focused on Leon County alone, removing large outlying areas of the MSA (PUMA \texttt{1206300}). Agents\footnote{Agents less than 20 years old were not include in the Foursquare POI data, thus they are excluded from our analysis.} were generated on the Federal Emergency Management Agency's (FEMA) open USA Structures database \cite{yang2018building, usa_struct_2022}, based on the sampling of synthetic households to single-family residential, multi-family residential, mobile homes, and group quarters housing types.
```

jvtcl replied:
- [ ] @jGaboardi to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\input tables/agents_counts_travel_mode
+
+Figure~\ref{fig:2x2_agents_hex} shows that Leon County's agent population is distributed unevenly relative to assigned travel modes. Because Leon County's infrastructure primarily supports travel by car, agents who drive are distributed closest to the area's general population density. The spatial distribution of agents who travel by walking also tends to follow Leon County's settlement patterns, ...
```
The readability and analysis of Figure 4 could be improved by moving it to page 3.

jvtcl replied:
- [ ] @jGaboardi to address this.

jGaboardi replied:
After we address and resolve all text edits, etc. we will attempt to accommodate this suggestion. However, the typesetting is done within LaTeX and we have limited control over the resultant PDF output.

ameyxd replied:
You may be able to force positions. It's okay if not. 👍🏽

jGaboardi replied:
I spent some time trying to manually place the figures as suggested, but in doing so they both (a) didn't fit properly in the desired space; and (b) completely threw off the typesetting of the other figures & tables. We will leave as is.

* @jvtcl Please mark complete

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+Grocery store POIs with medium to high visit confidence (at least 30 device visits per month, $n$ = 53) were obtained from Foursquare for Leon County in January 2023. Destination capacities were estimated based on visit counts weighted by representativeness of the demographic cohort within the state's 2010 Census population\footnote{https://location.foursquare.com/visits/docs/foursquare-data-norm ...
+
+After travel mode was assigned, four network cost matrices were calculated from origins (agents' residential locations) to destination POIs in \texttt{movelike}. Agents were then allocated to a single probable destination with the \texttt{actlike.ActivityAllocation} routine, which solves a modified Transportation Problem \cite{tp_miller_gc__2015}, where destination POI capacities are scaled \cite ...
```
Please elaborate on the original Transportation Problem. Is it based on the GIS-T? The referenced citation is broad.

jvtcl replied:
- [x] @jGaboardi to address this.

jGaboardi replied:
* I have added several more (and clearer) citation within our text and added some clarifying language.
* Here we are referring to and expanding on the [Hitchcock/Koopmans Transportation Problem](https://en.wikipedia.org/wiki/Transportation_theory_(mathematics)#Hitchcock_problem).
* @jvtcl Please mark as completed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+    \item A virtual machine (Ubuntu) with a 2.8 GHz 22-core Intel(R) Xeon(R) processor (86 GB RAM).
+\end{itemize}
+    
+The large disparity in problem size seen in Table~\ref{tab:agents_counts_travel_mode} is even more pronounced in solution runtimes, shown in Table~\ref{tab:allocation_solution_runtime}, with optimal solutions for non-drive models being found in a maximum time of just over 1 minute on both machines and the drive model taking more than 17 and 8 hours to solve on the macOS and Ubuntu machines, respe ...
```
In line with the suggestion to rearrange figures, consider showing Table 1 and Table 2 on the same page to highlight clear correlations in their content (high agent count corresponding to high runtime).

jvtcl replied:
- [ ] @jGaboardi to address this.

jGaboardi replied:
As [commented above](https://github.com/scipy-conference/scipy_proceedings/pull/834#discussion_r1242936515), we will attempt to accommodate this suggestion.

jGaboardi replied:
See comment above --> https://github.com/scipy-conference/scipy_proceedings/pull/834#discussion_r1256470229

* @jvtcl Please mark complete

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+To enable comparison with the results of \cite{likeness-scipy-paper-2022} and \cite{urbanpop-AG-2023}, which focused on anchor activities (work/school travel), we validated grocery store visits using Canonical Correlation Analysis (CCA) on the between-destination (relative prevalence) and within-destination (compositional) characteristics of each POI by demographic group. Both CCA runs were gener ...
+
+We also investigated local (POI-specific) within-destination correspondence in compositional characteristics based on gender/age cohorts in five-year intervals (until age 70, which is top-coded), in this case based on Spearman rank correlation due to the relatively small number of demographic cohorts ($n$ = 11).
```
Please cite Spearman correlation here.

jvtcl replied:
- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\input figures/local_demographic_validation.tex
+
+Each of the three P-MEDM runs (one for each PUMA in the Tallahassee CBSA) resulted in a synthetic reconstruction of the ACS 90\% MOEs that conformed within 99\% of the published MOEs from the ACS SF (Table~\ref{tab:demographic-validation}. Mapped to Leon County block groups (Figure~\ref{fig:local-demographic-validation}), MOE Fit Rates were somewhat lower but still conform within 90\% of the ACS  ...
```
somewhat -> use more formal words like 'relatively' or 'moderately'

jvtcl replied:
- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\input figures/local_demographic_validation.tex
+
+Each of the three P-MEDM runs (one for each PUMA in the Tallahassee CBSA) resulted in a synthetic reconstruction of the ACS 90\% MOEs that conformed within 99\% of the published MOEs from the ACS SF (Table~\ref{tab:demographic-validation}. Mapped to Leon County block groups (Figure~\ref{fig:local-demographic-validation}), MOE Fit Rates were somewhat lower but still conform within 90\% of the ACS  ...
```
Is there determined reasoning for why large population centers exhibit diminished performance? Can you mention that after concluding this sentence?

jvtcl replied:
- [x] @jvtcl to address this.

jvtcl replied:
> * @jvtcl to address this.

We feel that we don't have enough evidence from this single case study to provide a well-defined reason for why this occurs. A few potential reasons (population density, diversity of travel modes) are now provided in the text. We could explore this further in future work by repeating the case study analysis elsewhere.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+We found that overall, Likeness approximates travel to non-anchor (grocery store) POIs modestly well. However, contrasted with anchor activities (work/school) demonstrated in \cite{likeness-scipy-paper-2022}, non-anchor activities, like grocery store visits, occur at locations that do not have defined capacities (e.g., employee or enrollment sizes) are a greater challenge to represent realistical ...
+
+\begin{itemize}
+    \item The clearest \textbf{data-specific} challenge is temporal mismatch between the synthetic population (2019) and POI visit data (2023). We have yet to increment our ACS year due to sporadic issues of non-conforming geography between the 2010 PUMAs and 2020 block groups/tracts that we hope to resolve starting with the forthcoming 2022 ACS releases, which will begin to use 2020 PUMAs\footno ...
```
Reason for emboldening 'data-specific' and 'model-specific'?

jvtcl replied:
@ameyxd We emboldened this text to highlight the subject of each item. We could also place this text before the descriptions if you think that would be clearer.

jvtcl replied:
- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\subsection{Limitations}\label{section:limitations}
+
+We found that overall, Likeness approximates travel to non-anchor (grocery store) POIs modestly well. However, contrasted with anchor activities (work/school) demonstrated in \cite{likeness-scipy-paper-2022}, non-anchor activities, like grocery store visits, occur at locations that do not have defined capacities (e.g., employee or enrollment sizes) are a greater challenge to represent realistical ...
```
confounded -> more formal 'negatively impacted' or 'negatively influenced'

jvtcl replied:
- [x] @jvtcl to address this.

jvtcl replied:
@ameyxd We are in favor of leaving the wording here as-is, since we are identifying [confounding factors](https://www.sciencedirect.com/topics/nursing-and-health-professions/confounding-variable) associated with the inputs to our integrated model.

jvtcl replied:
Ended up revising this sentence to: "Multiple confounding factors related to data inputs and model assumptions may have affected our results..."

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\begin{itemize}
+    \item The clearest \textbf{data-specific} challenge is temporal mismatch between the synthetic population (2019) and POI visit data (2023). We have yet to increment our ACS year due to sporadic issues of non-conforming geography between the 2010 PUMAs and 2020 block groups/tracts that we hope to resolve starting with the forthcoming 2022 ACS releases, which will begin to use 2020 PUMAs\footno ...
+    \item Several large \textbf{model-specific} assumptions were made that could confound our results, particularly that 1) agents simultaneously travel to grocery stores, 2) agents only select one grocery store to visit, using only one mode of travel, and 3) travel to grocery stores only occurs between home and work. These assumptions can be updated by incorporating information from time-use and ...
+    \item Going forward, we also hope to tighten our assumptions about the feasibility of POI access relative to the various travel modes. For example, in our current assignment process (Figure~\ref{fig:mode_assign_workflow}) agents who we could not label with a defined travel mode were considered walkers as a fallback for unavailability of a bus route in their block group of residence. Future it ...
```
I do not see the purpose of having point 3 in this list, since it talks about potential changes you hope to make in future iterations. Consider having this as a conclusion of this section instead of a point.

jvtcl replied:
@ameyxd We agree with this suggestion and will make the recommended changes. 

- [ ] ~~@jvtcl~~ @jGaboardi to address this.

jvtcl replied:
To address @ameyxd's comment below, I added a few research directions we could investigate, one of them being "behavioral factors (willingness to travel given cost and impedances)". 

I moved the paragraph to the conclusion as recommended, but it feels redundant with this point. @jGaboardi, would you mind looking this over?

jGaboardi replied:
We have reworded and the adjusted text is now up in this PR.

* @jvtcl Please mark complete

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+\subsection{Conclusion and Outlook}\label{section:conclusion-outlook}
+
+This paper presented enhancements and scaling for the Likeness spatial microsimulation toolkit -- including implementing batched population synthesis runs and large-scale transportation network generation -- to enable activity modeling with customizable agent populations based on real-world housing stock and transportation infrastructure for any metropolitan statistical area (MSA) in the United S ...
```
The concluding statement is a bit vague. Please briefly point out one or two research questions that are evident from your initial exercise, and refrain from mentioning 'plan to tackle as our workflow develops' -> 'plan to answer in future work'.

jvtcl replied:
- [x] @jvtcl to address this.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+    \item \textbf{Open-sourcing core packages.} Though we have yet to meet our goal of open-sourcing the suite of Likeness packages \cite{likeness-scipy-paper-2022}, we are still on track to release the core packages for residential population synthesis, \texttt{pymedm} and \texttt{livelike}, in 2023, with releases of \texttt{movelike} and \texttt{actlike} likely to follow in 2024.
+    \item \textbf{Packaging schema.} A further consideration related to open-sourcing is whether we should migrate from a confederated toolkit schema where each module is a semi-independent Python package, as is seen in the modern implementation of PySAL \cite{pysal_GA_2022}, to a single monolithic Python package with submodules. Each schema has benefits and this decision will require much consid ...
+    \item \textbf{Consolidating visualization functionality.} We are in the process of consolidating functionality related to the visualization of input, processing, and results that have been used in an ad-hoc manner in the past. An initial push will be made for the inclusion of ``made-to-order'' population density hexbin plots and network-space allocation routes.
+    \item \textbf{Improving mobility modeling.} Realism in mobility modeling is a key area in which we are constantly striving for improvements, specifically transit. As stated previously in \nameref{section:likeness-expansion-scaling}, there is significant potential in exploring further integration of GTFS data for locally-accurate modeling.
```
Too subjective/informal, please rephrase the following: "Realism in mobility modeling is a key area in which we are constantly striving for improvements, specifically transit."

jvtcl replied:
- [x] @jGaboardi  to address this.

jGaboardi replied:
* I have revised that sentence.
* @jvtcl Please mark as completed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+    \item \textbf{Packaging schema.} A further consideration related to open-sourcing is whether we should migrate from a confederated toolkit schema where each module is a semi-independent Python package, as is seen in the modern implementation of PySAL \cite{pysal_GA_2022}, to a single monolithic Python package with submodules. Each schema has benefits and this decision will require much consid ...
+    \item \textbf{Consolidating visualization functionality.} We are in the process of consolidating functionality related to the visualization of input, processing, and results that have been used in an ad-hoc manner in the past. An initial push will be made for the inclusion of ``made-to-order'' population density hexbin plots and network-space allocation routes.
+    \item \textbf{Improving mobility modeling.} Realism in mobility modeling is a key area in which we are constantly striving for improvements, specifically transit. As stated previously in \nameref{section:likeness-expansion-scaling}, there is significant potential in exploring further integration of GTFS data for locally-accurate modeling.
+    \item \textbf{Optimization bottleneck.} As demonstrated in \nameref{section:results}, there is a clear hit in computational performance and runtime when solving  \texttt{actlike.ActivityAllocation} problems on increasingly larger model instances (e.g., more agents and more POIs). There are two paths to resolving this issue (which may be considered in concert): 1) Reviewing our modified Transp ...
```
Scalability issues are pretty common in simulation problems. I would like to see an additional sentence here, summarizing it and the general plan.

jvtcl replied:
- [ ] @jvtcl to discuss tooling for generating synthetic populations with national coverage.
- [x] @jGaboardi to discuss challenges associated with scaling activity allocation and ways forward. Benefits of an engine like `HiGHS` towards this task?

jGaboardi replied:
@jvtcl I have added some more explanatory language in for this. Please mark my portion as completed.

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+
+% bibliography
+\bibliographystyle{ieeetr}
+\bibliography{ mybib }
```
Overall, this is solid work, with quality content and descriptive diagrams/tables. It would benefit from the addition of some broader implications, insights, and details of the findings. Good job!

ameyxd commented on papers/james_gaboardi/paper.tex:
```diff
@@ -0,0 +1,129 @@
+\input figures/parse-to-attribution
+
+Agent generation at residential locations via \texttt{livelike} provides origin points for simulating travel to essential activities. 
+The next stage in the workflow employs network analysis to model the cost of travel to these activities \cite{OpenStreetMap, osmnx_CEUS_2017, foti_generalized_2012} and allocate agents to POIs accordingly based on mathematical programming routines \cite{mitchell_pulp_2011, santos_mixed_2020, lougee_coin_2003, forrest_coinorcbc_2023}. In our first iteration of Likeness, both these tasks were accom ...
```
Please cite OpenStreetMap for all figures outside of footnotes.

jvtcl replied:
@ameyxd Could you please clarify this point? Do you mean adding attribution to captions for figures that use OSM?

- [ ] @jGaboardi  to address this.

jGaboardi replied:
@ameyxd All plots that use OSM are already attributed within the figure themselves. Are you suggesting another layer of attritbution/citation?

ameyxd replied:
I noticed that. You can disregard this - all OSM attributions in figures are good.

ameyxd reviewed (changes requested):
Overall, solid work with quality content. Hope to see revisions and review the next draft soon.

ameyxd commented:
@jGaboardi Good work. Please @ me when you make changes to review - I am not being notified for this thread 👍🏽

jGaboardi commented:
> @jGaboardi Good work. Please @ me when you make changes to review - I am not being notified for this thread 👍🏽

@ameyxd Thank you for your detailed review! @jvtcl and I will go over you comments and respond.

kalyan678 commented:
> @kalyan678 Thank you completing your review! Perhaps you forgot to post your actual review comments?

@jGaboardi Sorry, my bad! I think you should see my comments now. Can you please confirm?

jGaboardi commented:
> > @kalyan678 Thank you completing your review! Perhaps you forgot to post your actual review comments?
> 
> @jGaboardi Sorry, my bad! I think you should see my comments now. Can you please confirm?

Yes, I can see. Thanks!

kalyan678 commented:
> > > @kalyan678 Thank you completing your review! Perhaps you forgot to post your actual review comments?
> > 
> > 
> > @jGaboardi Sorry, my bad! I think you should see my comments now. Can you please confirm?
> 
> Yes, I can see. Thanks!

@jGaboardi  Awesome!! Kindly @.. mention me when you make the modifications so that it will be easy for me to follow. Thanks!!

jGaboardi commented:
@ameyxd & @kalyan678, thanks so much for you detailed reviews. If you are both finished with comments at this time @jvtcl and I will go over them and respond in turn.

jGaboardi committed:
Update papers/james_gaboardi/paper.tex

Co-authored-by: Amey Ambade <ameyambade@gmail.com>

jGaboardi committed:
Update papers/james_gaboardi/paper.tex

Co-authored-by: Amey Ambade <ameyambade@gmail.com>

jGaboardi committed:
Update papers/james_gaboardi/paper.tex

Co-authored-by: Amey Ambade <ameyambade@gmail.com>

mepa commented:
@scoobies mark pending comment

Commit:
remove ecosystem png

Commit:
update figures

Commit:
update bib

Commit:
update metadata

Commit:
update paper

Commit:
rename ecosystem path

jGaboardi commented:
@ameyxd & @kalyan678 

We (@jvtcl and I) have addressed your comments and implemented your suggestions where possible. Thank you both for the thorough and thoughtful reviews. Your efforts have made this manuscript a much better piece of scholarship.

jGaboardi commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.5281/zenodo.7843975 is OK
- 10.5281/zenodo.7768810 is OK
- 10.1162/0899766042321814 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1002/sapm1941201224 is OK
- 10.1016/j.tbs.2015.03.002 is OK
- 10.2307/1907301 is OK
- 10.1147/rd.471.0057 is OK
- 10.1016/j.compenvurbsys.2013.03.004 is OK
- 10.1111/gec3.12204 is OK
- 10.1038/ncomms9166 is OK
- 10.25080/majora-212e5952-014 is OK
- 10.25080/majora-212e5952-02d is OK
- 10.6084/m9.figshare.c.6139131.v2 is OK
- 10.1007/s12061-015-9138-2 is OK
- 10.1080/01621459.1972.10481251 is OK

MISSING DOIs

- 10.1016/j.epidem.2022.100632 may be a valid DOI for title: Assessing K-12 school reopenings under different COVID-19 Spread scenarios–United States, school year 2020/21: A retrospective modeling study
- 10.1109/wsc.2018.8632409 may be a valid DOI for title: Chisim: An agent-based simulation model of social interactions in a large urban area
- 10.1109/jstars.2018.2835377 may be a valid DOI for title: Building extraction at scale using convolutional neural network: Mapping of the United States

INVALID DOIs

- https://doi.org/10.1111/gean.12276 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.apgeog.2022.102844 is INVALID because of 'https://doi.org/' prefix
- 10.2172/1868508 is INVALID
```

Commit:
rework several reference DOIs

jGaboardi commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.5281/zenodo.7843975 is OK
- 10.5281/zenodo.7768810 is OK
- 10.1016/j.epidem.2022.100632 is OK
- 10.1162/0899766042321814 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1002/sapm1941201224 is OK
- 10.1016/j.tbs.2015.03.002 is OK
- 10.2307/1907301 is OK
- 10.1147/rd.471.0057 is OK
- 10.1016/j.compenvurbsys.2013.03.004 is OK
- 10.1109/wsc.2018.8632409 is OK
- 10.1111/gec3.12204 is OK
- 10.1038/ncomms9166 is OK
- 10.25080/majora-212e5952-014 is OK
- 10.25080/majora-212e5952-02d is OK
- 10.1109/jstars.2018.2835377 is OK
- 10.6084/m9.figshare.c.6139131.v2 is OK
- 10.1007/s12061-015-9138-2 is OK
- 10.1080/01621459.1972.10481251 is OK

MISSING DOIs

- None

INVALID DOIs

- doi.org/10.1111/gean.12276 is INVALID because of 'doi.org/' prefix
- doi.org/10.1016/j.apgeog.2022.102844 is INVALID because of 'doi.org/' prefix
- 10.2172/1868508 is INVALID
```

Commit:
rework several reference DOIs [2]

jGaboardi commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.5281/zenodo.7843975 is OK
- 10.5281/zenodo.7768810 is OK
- 10.1016/j.epidem.2022.100632 is OK
- 10.1162/0899766042321814 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1002/sapm1941201224 is OK
- 10.1016/j.tbs.2015.03.002 is OK
- 10.2307/1907301 is OK
- 10.1147/rd.471.0057 is OK
- 10.1016/j.compenvurbsys.2013.03.004 is OK
- 10.1109/wsc.2018.8632409 is OK
- 10.1111/gec3.12204 is OK
- 10.1038/ncomms9166 is OK
- 10.1111/gean.12276 is OK
- 10.25080/majora-212e5952-014 is OK
- 10.25080/majora-212e5952-02d is OK
- 10.1016/j.apgeog.2022.102844 is OK
- 10.1109/jstars.2018.2835377 is OK
- 10.6084/m9.figshare.c.6139131.v2 is OK
- 10.1007/s12061-015-9138-2 is OK
- 10.1080/01621459.1972.10481251 is OK

MISSING DOIs

- None

INVALID DOIs

- 10.2172/1868508 is INVALID
```

ameyxd reviewed (approved):
Looks good. Approved! Hope to see you this week @jGaboardi @jvtcl @mepa

kalyan678 reviewed (approved):
@jGaboardi @jvtcl - I'm glad to hear that you found my comments and suggestions helpful. I have reviewed the changes made to the paper, and it appears satisfactory to me. I wish you the best of luck with your paper and future endeavors. Perhaps we will have the opportunity to collaborate again in the future. Enjoy the conference, and although I'll be attending virtually, I hope we can connect and network. Cheers!!
 @scoobies @deniederhut @mepa - I would like to express my sincere gratitude to the SciPy Team for providing this opportunity!!

mepa commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.compenvurbsys.2017.05.004 is OK
- 10.5281/zenodo.7843975 is OK
- 10.5281/zenodo.7768810 is OK
- 10.1016/j.epidem.2022.100632 is OK
- 10.1162/0899766042321814 is OK
- 10.1007/s12532-017-0130-5 is OK
- 10.1002/sapm1941201224 is OK
- 10.1016/j.tbs.2015.03.002 is OK
- 10.2307/1907301 is OK
- 10.1147/rd.471.0057 is OK
- 10.1016/j.compenvurbsys.2013.03.004 is OK
- 10.1109/wsc.2018.8632409 is OK
- 10.1111/gec3.12204 is OK
- 10.1038/ncomms9166 is OK
- 10.1111/gean.12276 is OK
- 10.25080/majora-212e5952-014 is OK
- 10.25080/majora-212e5952-02d is OK
- 10.1016/j.apgeog.2022.102844 is OK
- 10.1109/jstars.2018.2835377 is OK
- 10.6084/m9.figshare.c.6139131.v2 is OK
- 10.1007/s12061-015-9138-2 is OK
- 10.1080/01621459.1972.10481251 is OK

MISSING DOIs

- None

INVALID DOIs

- 10.2172/1868508 is INVALID
```

mepa commented:
Hey @jGaboardi, could you please check and resolve the INVALID DOI found by scoobies above? It means that `10.2172/1868508` wasn't found on https://doi.org/.

jGaboardi commented:
> Hey @jGaboardi, could you please check and resolve the INVALID DOI found by scoobies above? It means that `10.2172/1868508` wasn't found on https://doi.org/.

Yeah, this was giving us trouble, but it's a real document and the correct DOI. See [thread above](https://github.com/scipy-conference/scipy_proceedings/pull/834#issuecomment-1579057790) which links to [this document](https://science.osti.gov/-/media/Initiatives/pdf/Bioprepardeness_Roundtable_Report_092722.pdf) – DOI is listed on the second page there.

mepa commented:
> > Hey @jGaboardi, could you please check and resolve the INVALID DOI found by scoobies above? It means that `10.2172/1868508` wasn't found on https://doi.org/.
> 
> Yeah, this was giving us trouble, but it's a real document and the correct DOI. See [thread above](https://github.com/scipy-conference/scipy_proceedings/pull/834#issuecomment-1579057790) which links to [this document](https://science.osti.gov/-/media/Initiatives/pdf/Bioprepardeness_Roundtable_Report_092722.pdf) – DOI is listed on the second page there.

Great, thanks for the links!

mepa commented:
@scoobies mark ready

deniederhut reviewed (approved):
Thanks @ameyxd and @kalyan678 for helping us review this paper for the conference!

![Thank-You-gif-Free-882682437](https://github.com/scipy-conference/scipy_proceedings/assets/6868396/2fa5aea9-d55c-4d13-beac-6733e9118a7b)

jGaboardi commented:
Thanks again everyone for all the great work. Without this time and effort our community wouldn't exist.

kalyan678 commented:
> Thanks @ameyxd and @kalyan678 for helping us review this paper for the conference!
> 
> ![Thank-You-gif-Free-882682437](https://user-images.githubusercontent.com/6868396/258611124-2fa5aea9-d55c-4d13-beac-6733e9118a7b.gif)

Thank you again for the opportunity. I look forward to contributing much more in the coming days!

Cheers Everyone :)

ameyxd commented:
It was great working with you, and nice to meet you @jGaboardi and @jvtcl! 

@deniederhut hope to see you around town in Houston, and happy to help for next year. ✌️

deniederhut commented:
I politely request that we wait until it cools down a bit before grabbing a beer 😅. And yes, @jGaboardi always has great stuff to share!
